src/ert/scheduler/scheduler.py

```diff
@@ -319,19 +239,9 @@ async def _monitor_and_handle_tasks(
                         raise task_exception
 
             if not self.is_active():
-                if self._ee_uri is not None:
-                    try:
-                        await self._events.put(CLOSE_PUBLISHER_SENTINEL)
-                        await asyncio.wait_for(
-                            self._publisher_done.wait(), timeout=self._queue_timeout
-                        )
-                        await asyncio.wait_for(
-                            self._events.join(), timeout=self._queue_timeout
-                        )
-                    except asyncio.TimeoutError:
-                        logger.error(
-                            f"{self._events.qsize()} items left unprocessed in the queue!"
-                        )
+                if self._ee_queue:
+                    # only join queue if there is a consumer
```

Comment:
I find "join" to be a strange word for what is happening here, so in a comment I would write something about waiting.

Reply:
will do :+1: